auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/UmsClient.java

```diff
@@ -1156,6 +1151,37 @@ public GetUserSyncStateModelResponse getUserSyncStateModel(
         return newStub(requestId).getUserSyncStateModel(request);
     }
 
+    public Set<String> listResourceRoles(String requestId, String accountId) {
+        UserManagementProto.ListResourceRolesResponse response = newStub(requestId).listResourceRoles(UserManagementProto.ListResourceRolesRequest.newBuilder()
+                .setAccountId(accountId)
+                .build());
+        Set<String> resourceRoles = Sets.newHashSet();
+        resourceRoles.addAll(response.getResourceRoleList().stream().map(resourceRole -> resourceRole.getCrn()).collect(Collectors.toSet()));
+        while (response.hasNextPageToken()) {
+            response = newStub(requestId).listResourceRoles(UserManagementProto.ListResourceRolesRequest.newBuilder()
```

Comment:
Please extract the stub into a local variable and not create a new one every time in a loop.